src/hooks/useDSHeading.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+
+import Heading from "../components/Heading/Heading";
+
+interface UseDSHeadingProps {
+  title: string | JSX.Element;
+  id?: string;
+  customDefaultHeading?: string | JSX.Element;
+  additionalStyles?: { [key: string]: any };
+}
+
+const headingList = ["h1", "h2", "h3", "h4", "h5", "h6"];
+
+/**
+ * DS internal helper hook to render a default `h2` heading element if the
+ * passed title is a string. Otherwise, it will return the title as is if it
+ * is a JSX element.
+ */
+function useDSHeading({
+  title = "",
+  id = "",
+  customDefaultHeading,
+  additionalStyles,
+}: UseDSHeadingProps) {
+  const headingID = id ? `${id}-heading` : undefined;
+  let updatedTitle = null;
+
+  if (title) {
+    if (typeof title === "string") {
+      // Use the DS default h2 heading element if the title is a string.
+      updatedTitle = (
+        <Heading id={headingID} __css={additionalStyles}>
+          {title}
+        </Heading>
+      );
+    } else if (typeof title === "object") {
+      // Use the passed JSX element if it is passed. If additional styles are
+      // passed, clone the element and add the styles to the cloned element.
+      // This is to account for base default styles for a component that may
+      // be ignored in a custom heading.
+      if (title.type === Heading) {
+        updatedTitle = additionalStyles
+          ? React.cloneElement(title, { __css: additionalStyles })
+          : title;
+      } else if (headingList.includes(title.type)) {
+        // If the user passed in an HTML heading element, just use that.
+        // They will not get any DS styling.
+        updatedTitle = title;
+        console.warn(
```

Comment:
@jackiequach @bigfishdesign13 I am logging two warnings:
1 - one where an HTML heading element is passed,
2 - one for non-DS `Heading` or HTML heading elements are passed.

The wording could be better. What do you think? Also added tests for these.

Reply:
These are very relevant warning messages.  I just made some suggetions for the wording.